src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -0,0 +1,148 @@
+import { Column } from 'react-virtualized/dist/es/Table/index';
+
+class ArchiveExplorer extends React.Component {
+    static propTypes = {
+        data: PropTypes.arrayOf(
+            PropTypes.shape({
+                type: PropTypes.string,
+                absolute_path: PropTypes.string,
+                name: PropTypes.string,
+                modified_at: PropTypes.string,
+                size: PropTypes.number,
+                path_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string,
+                            name: PropTypes.string,
+                        }),
+                    ),
+                }),
+                parent: PropTypes.string,
+                item_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string,
+                            name: PropTypes.string,
+                        }),
+                    ),
+                }),
+            }),
+        ).isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        const { cache, root } = this.buildCache(props.data);
+        this.cache = cache;
+        this.root = root;
+
+        this.state = {
+            currentData: this.getCurrentData(cache, root),
+        };
+    }
+
+    buildCache = data => {
+        const cache = {};
+        let root;
+
+        data.forEach(info => {
+            if (!info.parent) {
+                root = info.absolute_path;
+            }
+            cache[info.absolute_path] = info;
+        });
+
+        return { cache, root };
+    };
+
+    getCurrentData = (cache, currentFolder) => {
+        const currentFolderInfo = cache[currentFolder];
+
+        const currentData = currentFolderInfo.item_collection.entries.map(item => cache[item.absolute_path]);
+
+        return currentData;
+    };
+
+    updateData = absolutePath => {
+        this.setState({
+            currentData: this.getCurrentData(this.cache, absolutePath),
+        });
+    };
+
+    getRowData = ({ index }) => {
+        const { currentData } = this.state;
+        const currentRow = currentData[index];
+        const { name, type, modified_at: modifiedAt, absolute_path: absolutePath } = currentRow;
+
+        const rowData = {
+            ...currentRow,
+            name: {
+                id: undefined,
+                name,
+                isExternal: false,
+                type,
+                onClick: type === 'folder' ? () => this.updateData(absolutePath) : undefined,
+            },
+            date: `20${modifiedAt}`,
```

Comment:
This doesn't seem ideal. Are we expecting full date objects in the final data structure?

Reply:
This is only for now, because it's not standard date format and JS cannot parse it correctly. I talked to conversion team, and they said they would fix it.